### PR TITLE
Xr Scan commands - Fix Previous PR Comments

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: xray/cli.go [hunk 1]
```diff
@@ -274,6 +285,23 @@ func createServerDetailsWithConfigOffer(c *cli.Context) (*coreconfig.ServerDetai
 }
 
 func shouldIncludeVulnerabilities(c *cli.Context) bool {
-	// If no contex was provided by the user, no Violations will be triggered by Xray, so include general vulnerabilities in the command output
+	// If no context was provided by the user, no Violations will be triggered by Xray, so include general vulnerabilities in the command output
 	return c.String("watches") == "" && c.String("project") == "" && c.String("repo-path") == ""
 }
+
+func validateXrayContex(c *cli.Context) error {
```
Comment (yahavi) [inline xray/cli.go:292]:
```suggestion
func validateXrayContext(c *cli.Context) error {
```

File: utils/cliutils/commandsflags.go [hunk 2]
```diff
@@ -1032,7 +1032,7 @@ var flagsMap = map[string]cli.Flag{
 	},
 	xrayScan: cli.StringFlag{
 		Name:  xrayScan,
-		Usage: "[Default: false]  Set to true if you'd like artifacts upload only after passing successfully an Xray scan.` `",
+		Usage: "[Default: false] Set to true if you'd like artifacts upload only after passing successfully an Xray scan.` `",
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:1035]: Set to true if you'd like artifacts upload only after passing successfully an Xray scan.
-->
Set if you'd like all files to be scanned by Xray on the local file system prior to the upload, and skip the upload if any of the files are found vulnerable.

File: utils/cliutils/commandsflags.go [hunk 3]
```diff
@@ -1065,11 +1065,11 @@ var flagsMap = map[string]cli.Flag{
 	},
 	watches: cli.StringFlag{
 		Name:  watches,
-		Usage: "[Optional] A comma seperated list of Xray's watches, that will determine Xray's violations creation. ` `",
+		Usage: "[Optional] A comma seperated list of Xray watches, that to determine Xray's violations creation. ` `",
```
Comment (eyalbe4) [inline utils/cliutils/commandsflags.go:1068]: A comma seperated list of Xray watches, that to determine Xray's violations creation.
-->
A comma separated list of Xray watches, to determine Xray's violations creation.

File: xray/cli.go [hunk 4]
```diff
@@ -274,6 +285,23 @@ func createServerDetailsWithConfigOffer(c *cli.Context) (*coreconfig.ServerDetai
...
+	contextFound := false
+	contextErorrMsg := "only one of the following flags can be supplied: --watches,--project or --repo-path."
```
Comment (eyalbe4) [inline xray/cli.go:294]: Missing white-space after ```--watches,```. Should be:
...supplied: --watches, --project or --repo-path.

File: xray/cli.go [hunk 5]
```diff
@@ -274,6 +285,23 @@ func createServerDetailsWithConfigOffer(c *cli.Context) (*coreconfig.ServerDetai
...
+	if c.String("project") != "" && !contextFound {
+		contextFound = true
+	} else if c.String("project") != "" {
+		return errorutils.CheckError(errors.New(contextErorrMsg))
+	}
+	if c.String("repo-path") != "" && contextFound {
+		return errorutils.CheckError(errors.New(contextErorrMsg))
+	}
+	return nil
+}
```
Comment (eyalbe4) [inline xray/cli.go:307]: The above function can be simplified as follows:
```
func validateXrayContex(c *cli.Context) error {
	i := 0
	if c.String("watches") != "" {
		i++
	}
	if c.String("project") != "" {
		i++
	}
	if c.String("repo-path") != "" {
		i++
	}
	if u > 1 {
		return errorutils.CheckError(errors.New("only one of the following flags can be supplied: --watches,--project or --repo-path"))
	}
	return nil
}
```

Also, let's add a unit test for this function. See examples for testing flags in ```cli_test.go```.

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

**Pull request merged**